src/plugins/embeddable/public/embeddable_panel/panel_header/embeddable_panel_title.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import classNames from 'classnames';
+import React, { useMemo } from 'react';
+import { EuiIcon, EuiLink, EuiToolTip } from '@elastic/eui';
+
+import { IEmbeddable, ViewMode } from '../../lib';
+import { CustomizePanelAction } from '../panel_actions';
+import { useSelectFromEmbeddableInput } from '../use_select_from_embeddable';
+import { getEditTitleAriaLabel, placeholderTitle } from '../embeddable_panel_strings';
+
+export const EmbeddablePanelTitle = ({
+  hideTitle,
+  embeddable,
+  customizePanelAction,
+}: {
+  hideTitle?: boolean;
+  embeddable: IEmbeddable;
+  customizePanelAction?: CustomizePanelAction;
+}) => {
+  const title = embeddable.getTitle();
```

Comment:
Parent component `EmbeddablePanelHeader` also sets up subscriptions for `title`, `viewMode`, and `description`. How about just taking these values as props to avoid duplicating state and $input subscriptions here?

Reply:
I usually like having separate calls to `select` directly inside the components in which that state gets used. I'm not totally sure if this is a performance issue, but will do some research. I see prop drilling as a bit of a trade-off in that it _can_ occasionally lower readability.

Reply:
There is a best practice for this in the [redux style guide](https://redux.js.org/style-guide/) called Connect More Components to Read Data from the Store. This isn't exactly the same, because this is using a subscription from RXJS, but the selector uses some sort of subscription under the hood, so I think it might be _similar_ at least.